be/src/service/backend_base.h

```diff
@@ -64,6 +64,22 @@ class BackendServiceBase : public BackendServiceIf {
 
     ~BackendServiceBase() override = default;
 
+    // NOTE: now we do not support multiple backend in one process
+    template <class service>
+    static std::unique_ptr<ThriftServer> create(ExecEnv* exec_env, int port) {
```

Comment:
need to implement this function in cpp with template instantiation
this can avoid unnecessary include and dependency in header files.